dask/array/tests/test_xarray.py

```diff
@@ -22,3 +22,15 @@ def test_asanyarray():
     y = da.asanyarray(xr.DataArray([1, 2, 3.0]))
     assert isinstance(y, da.Array)
     assert_eq(y, y)
+
+
+def test_asarray_recognises_xarray_data():
+    module = xr.DataArray.__module__
+    try:
+        xr.DataArray.__module__ = "xarray"
+        y = da.asarray(xr.DataArray([1, 2, 3.0]))
+        assert isinstance(y, da.Array)
+        assert type(y._meta).__name__ == "ndarray"
```

Comment:
Just curious why this check is needed

Reply:
This is the assert that fails without the fix.